src/components/app.tsx

```diff
@@ -5,13 +5,43 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
+import {
+  ActionHandler,
+  DatasetAsyncAction,
+  datasetUrlLoad,
+} from '../actions';
+
+
+export type VoyagerConfig = Object;
+export type VoyagerData = InlineData | UrlData;
+
+interface VoyagerAppProps extends ActionHandler<DatasetAsyncAction> {
+  config: Object;
+  data: Data;
```

Comment:
i think that better separates out the concerns of whether you want an initial dataset to be something in particular vs ability to switch between things. I don't think the UI part for the selector should depend on the currently set data.